### ISSUE TITLE
16 of 28 gradient-based optimizers lack IFusedOptimizerSpec, silently disabling fused compiled training

### ISSUE THREAD

**Issue by ooples** [ooples/AiDotNet#1930]
## 16 of 28 gradient-based optimizers cannot use the fused compiled-training path

`CompiledTapeTrainingStep.TryStepWithFusedOptimizer` requires the optimizer to implement `Fused.IFusedOptimizerSpec`. Optimizers that do not are silently dropped to the eager autograd tape, so **the choice of optimizer alone decides whether a model gets the compile-once-replay-many fast path** — with no diagnostic surfaced to the user.

### Currently implemented (12)

`AdamOptimizer` · `AdamWOptimizer` · `AMSGradOptimizer` · `Adam8BitOptimizer` · `NadamOptimizer` · `AdaMaxOptimizer` · `AdaDeltaOptimizer` · `AdagradOptimizer` · `RootMeanSquarePropagationOptimizer` · `LAMBOptimizer` · `LionOptimizer` · `StochasticGradientDescentOptimizer`

### Missing (16)

**Group A — elementwise first-order, should be straightforward to add.** These are per-parameter update rules of the same shape the existing fused kernels already express, so the fused config is mostly a matter of mapping hyperparameters to an `OptimizerType`:

- [ ] `GradientDescentOptimizer`
- [ ] `MiniBatchGradientDescentOptimizer`
- [ ] `MomentumOptimizer`
- [ ] `NesterovAcceleratedGradientOptimizer`
- [ ] `LARSOptimizer`
- [ ] `FTRLOptimizer`
- [ ] `ProximalGradientDescentOptimizer`

Note `StochasticGradientDescentOptimizer` already has a spec, so plain and mini-batch GD being absent is an inconsistency rather than a design decision.

**Group B — not expressible as a fused elementwise update.** Second-order, quasi-Newton, line-search and splitting methods need curvature estimates, history buffers, or multiple function evaluations per step, none of which a per-element kernel can represent. These likely warrant an explicit "cannot fuse" marker plus a documented reason, rather than an implementation:

- [ ] `BFGSOptimizer` · `LBFGSOptimizer` · `DFPOptimizer`
- [ ] `NewtonMethodOptimizer` · `LevenbergMarquardtOptimizer`
- [ ] `TrustRegionOptimizer` · `ConjugateGradientOptimizer`
- [ ] `CoordinateDescentOptimizer` · `ADMMOptimizer`

### Why this matters beyond performance

This gap masked a correctness bug for a long time. `RWKVForecaster` produced NaN across its whole parameter buffer after a single training step, intermittently (~50% of runs). The investigation recorded that it "only reproduces with stateful optimizers", which was read as evidence of a memory-allocator fault caused by extra moment buffers.

The real explanation was this gap: the control arm used `GradientDescentOptimizer`, which has no fused spec, so compiled training never engaged for that arm at all. The fault was in fused compiled training over a recurrent forward — nothing to do with optimizer state. Fixed by opting RWKVForecaster out of `SupportsFusedCompiledTraining`, matching every sibling in its family (HiPPO, GatedDeltaNet, GLA, Griffin, Hawk).

So partial coverage does not just cost speed; it makes the fused path an invisible variable that confounds debugging. Two concrete asks fall out of that:

1. **Surface the miss reason.** `EmitFusedMissAndFallback` already produces one internally — making it observable (a counter, or a diagnostic hook) would have shortened that investigation considerably.
2. **Make coverage explicit.** Every gradient-based optimizer should either implement the spec or declare that it cannot and why, so "no fused spec" is never silently indistinguishable from "not yet done".

### Verification

```
$ for f in src/Optimizers/*.cs; do
    grep -qE "GradientBasedOptimizerBase<|: IGradientBasedOptimizer" "$f" \
      && ! grep -q "IFusedOptimizerSpec" "$f" && basename "$f";
  done
```

**Comment by ooples** [comment]
Confirmed still outstanding on the PR #1789 branch. `IFusedOptimizerSpec` is referenced by 14 files under `src/Optimizers/`, of which 2 are the interface itself and `GradientBasedOptimizerBase` — so the same ~12 concrete optimizers implement it, and none of the 16 listed here (GradientDescent, MiniBatchGradientDescent, Momentum, Nesterov, LARS, FTRL, ProximalGradientDescent, BFGS, LBFGS, DFP, Newton, LevenbergMarquardt, TrustRegion, ConjugateGradient, CoordinateDescent, ADMM) has gained it.
